### ISSUE TITLE
`[heap]` causes memory leak

### ISSUE THREAD

**Issue by kahsa** [vlang/v#11102]
**V version:  V 0.2.2 03ed32f** 
**OS: ArchLinux** 

**What did you do?**
```v
import time

[heap]  // removing [heap] fixes the leak
struct Foo{
    arr []string
}

fn (f Foo) size() int {
    return f.arr.len
}

fn main() {
    f := Foo{arr:[]string{len:100000, init: 'a'}}
    for {
        // n := f.arr.len // this does not leak even if with [heap]
        n := f.size()
        time.sleep(1 * time.millisecond)
    }
}
```

**What did you expect to see?**

 
**What did you see instead?**

heavy memory leaks.

**Comment by kahsa** [comment]
#10429  may be related.

**Comment by medvednikov** [comment]
Fixed in 0.3
